contracts/bridge/contracts/LightClientBridge.sol

```diff
@@ -198,189 +263,183 @@ contract LightClientBridge is Pausable, Initializable, ValidatorRegistry {
 
         // Accept and save the commitment
         validationData[currentId] = ValidationData(
+            msg.sender,
             commitmentHash,
-            validatorClaimsBitfield,
-            block.number
+            block.number,
+            validatorClaimsBitfield
         );
 
         emit InitialVerificationSuccessful(msg.sender, block.number, currentId);
 
-        currentId = currentId.add(1);
+        currentId = currentId + 1;
     }
 
     /**
      * @notice Performs the second step in the validation logic
      * @param id an identifying value generated in the previous transaction
      * @param commitment contains the full commitment that was used for the commitmentHash
-     * @param signatures an array of signatures from the randomly chosen validators
-     * @param validatorPositions an array of bitfields from the chosen validators
-     * @param validatorPublicKeys an array of the public key of each signer
-     * @param validatorPublicKeyMerkleProofs an array of merkle proofs from the chosen validators
+     * @param validatorProof a struct containing the data needed to verify all validator signatures
      */
     function completeSignatureCommitment(
         uint256 id,
         Commitment memory commitment,
-        bytes[] memory signatures,
-        uint256[] memory validatorPositions,
-        address[] memory validatorPublicKeys,
-        bytes32[][] memory validatorPublicKeyMerkleProofs
-    ) public whenNotPaused {
+        ValidatorProof memory validatorProof
+    ) public {
         // only current epoch
-        require(commitment.validatorSetId == validatorSetId, "Error: Invalid validator set id");
+        require(commitment.validatorSetId == validatorSetId, "Bridge: Invalid validator set id");
+
+        verifyCommitment(id, commitment, validatorProof);
+
+        processPayload(commitment.payload, commitment.blockNumber);
+
+        emit FinalVerificationSuccessful(msg.sender, id);
```

Comment:
suggest: move emit to the end of function.

Reply:
Done